samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/settings/SettingsActivity.java

```diff
@@ -192,4 +141,75 @@ private void updateIOWithConfig(@NonNull CustomerIOSDKConfig config) {
+            isFormValid = false;
+        } else {
+            ViewUtils.setError(binding.trackingUrlInputLayout, null);
+        }
+
+        String siteId = ViewUtils.getTextTrimmed(binding.siteIdTextInput);
+        if (TextUtils.isEmpty(siteId)) {
+            ViewUtils.setError(binding.siteIdInputLayout, getString(R.string.error_text_input_field_blank));
+            isFormValid = false;
+        } else {
+            ViewUtils.setError(binding.siteIdInputLayout, null);
+        }
+
+        String apiKey = ViewUtils.getTextTrimmed(binding.apiKeyTextInput);
+        if (TextUtils.isEmpty(apiKey)) {
+            ViewUtils.setError(binding.apiKeyInputLayout, getString(R.string.error_text_input_field_blank));
+            isFormValid = false;
+        } else {
+            ViewUtils.setError(binding.apiKeyInputLayout, null);
+        }
+
+        String bqSecondsDelayText = ViewUtils.getTextTrimmed(binding.bqDelayTextInput);
+        Double bqSecondsDelay = StringUtils.parseDouble(bqSecondsDelayText, null);
+        if (TextUtils.isEmpty(bqSecondsDelayText)) {
+            ViewUtils.setError(binding.bqDelayInputLayout, getString(R.string.error_text_input_field_blank));
+            isFormValid = false;
+        } else {
+            ViewUtils.setError(binding.bqDelayInputLayout, null);
+        }
+
+        String bqMinTasksText = ViewUtils.getTextTrimmed(binding.bqTasksTextInput);
+        Integer bqMinTasks = StringUtils.parseInteger(bqMinTasksText, null);
+        if (TextUtils.isEmpty(bqMinTasksText)) {
+            ViewUtils.setError(binding.bqTasksInputLayout, getString(R.string.error_text_input_field_blank));
+            isFormValid = false;
+        } else {
+            ViewUtils.setError(binding.bqTasksInputLayout, null);
+        }
+
+        if (isFormValid) {
+            binding.progressIndicator.show();
+            boolean featTrackScreens = binding.trackScreensSwitch.isChecked();
+            boolean featTrackDeviceAttributes = binding.trackDeviceAttributesSwitch.isChecked();
+            boolean featDebugMode = binding.debugModeSwitch.isChecked();
+            CustomerIOSDKConfig config = new CustomerIOSDKConfig(siteId,
+                    apiKey,
+                    trackingURL,
+                    bqSecondsDelay,
+                    bqMinTasks,
+                    featTrackScreens,
+                    featTrackDeviceAttributes,
+                    featDebugMode);
+            Disposable disposable = settingsViewModel
+                    .updateConfigurations(config)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(preferences -> {
+                        binding.progressIndicator.hide();
+                        Toast.makeText(this, R.string.settings_save_msg, Toast.LENGTH_SHORT).show();
+                        OSUtils.restartApp();
+                    });
+            disposables.add(disposable);
+        }
+    }
```

Comment:
I think there is a lot of duplicated code, can we combine them and make it easier? 

something like this? 


```
private boolean isValidForm() {
    boolean isFormValid = true;

    isFormValid &= validateAndSetError(binding.trackingUrlInputLayout, this::isTrackingURLValid, R.string.error_tracking_url);
    isFormValid &= validateAndSetError(binding.siteIdInputLayout, TextUtils::isEmpty, R.string.error_text_input_field_blank);
    isFormValid &= validateAndSetError(binding.apiKeyInputLayout, TextUtils::isEmpty, R.string.error_text_input_field_blank);
    isFormValid &= validateAndSetError(binding.bqDelayInputLayout, TextUtils::isEmpty, R.string.error_text_input_field_blank);
    isFormValid &= validateAndSetError(binding.bqTasksInputLayout, TextUtils::isEmpty, R.string.error_text_input_field_blank);

    return isFormValid;
}

private boolean validateAndSetError(TextInputLayout inputLayout, Predicate<String> predicate, int errorStringId) {
    String input = ViewUtils.getTextTrimmed(inputLayout.getEditText());

    if (predicate.test(input)) {
        ViewUtils.setError(inputLayout, getString(errorStringId));
        return false;
    } else {
        ViewUtils.setError(inputLayout, null);
        return true;
    }
}
```

Reply:
I had to make [some tweaks](https://github.com/customerio/customerio-android/pull/222/commits/bcc20b1f7d861dee3cdd243becb9f6f817f2c974), pushed after following similar approach